src/runtime.rs

```diff
@@ -0,0 +1,154 @@
+//! Tokio runtime
+
+use reactor::{self, Reactor};
+use reactor::background::{self, Background};
+
+use tokio_executor;
+use tokio_threadpool::{self as threadpool, ThreadPool};
+use futures::Poll;
+use futures::future::{Future, Join};
+
+use std::io;
+
+/// Handle to the Tokio runtime.
+///
+/// The Tokio runtime includes a reactor as well as an executor for running
+/// tasks.
+#[derive(Debug)]
+pub struct Runtime {
+    inner: Option<Inner>,
+}
+
+/// Error returned from `with_context`.
+#[derive(Debug)]
+pub struct EnterError(());
+
+/// A future that resolves when the Tokio `Runtime` is shut down.
+#[derive(Debug)]
+pub struct Shutdown {
+    inner: Join<background::Shutdown, threadpool::Shutdown>,
+}
+
+#[derive(Debug)]
+struct Inner {
+    /// Reactor running on a background thread.
+    reactor: Background,
+
+    /// Task execution pool.
+    pool: ThreadPool,
+}
+
+// ===== impl Runtime =====
+
+/// Start the Tokio runtime.
+pub fn run<F: FnOnce() -> R, R>(f: F) -> R {
+    let mut runtime = Runtime::new().unwrap();
+    let ret = runtime.with_context(f).unwrap();
+
+    runtime.shutdown().wait().unwrap();
+
+    ret
+}
+
+/// Start the Tokio runtime and spawn the provided future.
+pub fn run_seeded<F>(f: F)
```

Comment:
this is trivial but the 'seeded' terminology seems very unfamiliar for what's probably a common entrypoint to the API.  The only 'seed' that comes to mind is that of random number generation.  What about just 'run_future' as the name here?

Reply:
I agree that the name isn't *great*.

The issue with `run_future` is that the function does more than this. It spawns the future onto the runtime, runs it AND any other futures that get spawned.

Any other thoughts for names?

Reply:
`go` ? :trollface: 
How about `run_async` ?

Reply:
`run` for closure, `poll` for future?

Reply:
I'm thinking of just renaming `run_seeded` -> `run` and getting rid of the other fn (in favor of using the `Runtime` instance directly).